sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -87,6 +87,62 @@ case class MapKeys(child: Expression)
   override def prettyName: String = "map_keys"
 }
 
+@ExpressionDescription(
+  usage = """_FUNC_(a1, a2) - Returns a merged array matching N-th element of first
+  array with the N-th element of second.""",
+  examples = """
+    Examples
+      > SELECT _FUNC_(array(1, 2, 3), array(2, 3, 4));
+        [[1, 2], [2, 3], [3, 4]]
+  """,
+  since = "2.4.0")
+case class Zip(left: Expression, right: Expression)
+  extends BinaryExpression with ExpectsInputTypes {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, ArrayType)
+
+  override def dataType: DataType = ArrayType(left.dataType.asInstanceOf[ArrayType].elementType)
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    nullSafeCodeGen(ctx, ev, (arr1, arr2) => {
+      val i = ctx.freshName("i")
+      s"""
+      for (int $i = 0; $i < $arr1.numElements(); $i ++) {
+        if ($arr1.isNullAt($i)) {
+          ${ev.isNull} = true;
```

Comment:
Is this assignment for each element?

Reply:
Sorry, it is under WIP. Just a curious.

Reply:
Nah, it was a mistake - I should break I think. Thank you!